frontend/src/components/App/Home/index.tsx

```diff
@@ -89,14 +94,43 @@ function ContextMenu({ cluster }: { cluster: Cluster }) {
         {helpers.isElectron() && cluster.meta_data?.source === 'dynamic_cluster' && (
           <MenuItem
             onClick={() => {
-              removeCluster(cluster);
+              setOpenConfirmDialog(true);
               handleMenuClose();
             }}
           >
             <ListItemText>{t('translation|Delete')}</ListItemText>
           </MenuItem>
         )}
       </Menu>
+
+      <Dialog
```

Comment:
I think here you can use the `ConfirmDialog` component to make the UI consistent 
https://github.com/headlamp-k8s/headlamp/blob/main/frontend/src/components/common/ConfirmDialog.tsx

Reply:
Thank you, make sense. Updated